syftlib/src/test/java/org/openmined/syft/unit/monitor/network/NetworkStatusRealDataSourceTest.kt

```diff
@@ -0,0 +1,155 @@
+package org.openmined.syft.unit.monitor.network
+
+import android.accounts.NetworkErrorException
+import android.net.NetworkCapabilities
+import org.junit.Test
+import org.openmined.syft.common.AbstractSyftWorkerTest
+import org.openmined.syft.common.HttpClientMock
+import org.openmined.syft.common.SocketClientMock
+import org.openmined.syft.domain.SyftConfiguration
+import org.openmined.syft.monitor.network.NetworkStatusModel
+import org.openmined.syft.monitor.network.NetworkStatusRealTimeDataSource
+
+@ExperimentalUnsignedTypes
+class NetworkStatusRealDataSourceTest : AbstractSyftWorkerTest() {
+
+    private val workerId = "test id"
+    private val networkStatusModel = NetworkStatusModel()
+
+    private var networkDataSourceSuccess: NetworkStatusRealTimeDataSource
+    private var networkDataSourceFailure: NetworkStatusRealTimeDataSource
+
+    init {
```

Comment:
This could go in @Before setup() check syftJobTest

Reply:
Yeah, I saw it